packages/build/src/tarball.spec.ts

```diff
@@ -8,180 +8,131 @@ import os from 'os';
-      const expectedTarball = tarballPath(__dirname, BuildVariant.Redhat, version);
-
-      // Our ubuntu box, which for the time being runs this test, does not come
-      // with `rpmbuild` installed, so we end up getting:
-
-      // Error: spawn rpmbuild ENOENT
-
-      // Apt does not have `rpmbuild` package, so we will not able to install it
-      // on ubuntu's build variant. We can, however, run this test fully once we
-      // specifically test on rhel80-large. For now just skip if `rpmbuild` is
-      // not available.
-      beforeEach(function() {
-        if (!commandExists.sync('rpmbuild')) {
-          this.skip();
-        }
+    // macos and windows build variants does not come with installed 'dpkg' so
+    // this test fails. Run this test only on linux platforms.
+    if (os.platform() === Platform.Linux) {
+      describe('.tarballDebian', () => {
+        it('packages the executable(s)', async() => {
+          const tarball = await createTarball(tarballDir, 'debian', pkgConfig);
+          await fs.access(tarball.path);
+          {
+            const { stdout } = await execFile('dpkg', ['-c', tarball.path]);
+            expect(stdout).to.match(/^-rwxrwxr-x.+\/usr\/bin\/foo$/m);
+            expect(stdout).to.match(/^-rwxrwxr-x.+\/usr\/libexec\/bar$/m);
+            expect(stdout).to.match(/^-rw-rw-r--.+\/usr\/share\/doc\/foobar\/LICENSE_bar$/m);
+            expect(stdout).to.match(/^-rw-rw-r--.+\/usr\/share\/doc\/foobar\/LICENSE_foo$/m);
+            expect(stdout).to.match(/^-rw-rw-r--.+\/usr\/share\/doc\/foobar\/README$/m);
+            expect(stdout).to.match(/^-rw-rw-r--.+\/usr\/share\/doc\/foobar\/copyright$/m);
+          }
+          {
+            const { stdout } = await execFile('dpkg', ['-I', tarball.path]);
+            expect(stdout).to.match(/Package: foobar/);
+            expect(stdout).to.match(/Version: 1.0.0/);
+            expect(stdout).to.match(/Maintainer: Somebody <somebody@example.org>/);
+            expect(stdout).to.match(/Homepage: https:\/\/example.org/);
+            expect(stdout).to.match(/Description: Dummy package/);
+          }
+        });
       });
 
-      afterEach(async() => {
-        // only run this afterEach if we have rpmbuild command available, i.e.
-        // this test was not skipped.
-        if (commandExists.sync('rpmbuild')) {
-          await fs.unlink(expectedTarball);
-        }
+      describe('.tarballRedhat', () => {
+        // Our ubuntu box, which for the time being runs this test, does not come
+        // with `rpmbuild` installed, so we end up getting:
+
+        // Error: spawn rpmbuild ENOENT
+
+        // Apt does not have `rpmbuild` package, so we will not able to install it
+        // on ubuntu's build variant. We can, however, run this test fully once we
+        // specifically test on rhel80-large. For now just skip if `rpmbuild` is
+        // not available.
+        beforeEach(function() {
```

Comment:
I’m not sure, maybe? But `beforeEach()` makes sense to me, because that way it’s more obvious that every test gets skipped (even though it’s not really obvious anyway that skipping in a before hook skips the test itself, tbh)